protocols/gossipsub/src/protocol.rs

```diff
@@ -267,37 +267,28 @@ impl Encoder for GossipsubCodec {
     type Item = rpc_proto::Rpc;
     type Error = GossipsubHandlerError;
 
-    fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error> {
-        let mut buf = Vec::with_capacity(item.encoded_len());
-
-        item.encode(&mut buf)
-            .expect("Buffer has sufficient capacity");
-
-        // length prefix the protobuf message, ensuring the max limit is not hit
-        self.length_codec
-            .encode(Bytes::from(buf), dst)
-            .map_err(|_| GossipsubHandlerError::MaxTransmissionSize)
+    fn encode(
+        &mut self,
+        item: Self::Item,
+        dst: &mut BytesMut,
+    ) -> Result<(), GossipsubHandlerError> {
+        match self.codec.encode(item, dst) {
+            Ok(_) => Ok(()),
+            _ => Err(GossipsubHandlerError::MaxTransmissionSize),
+        }
```

Comment:
```suggestion
        match self.codec.encode(item, dst) {
           return Ok(self.codec.encode(item, dst)?);
        }
```
Ok, after quickly reading up on `?` and conversing with the compiler, I think that change is what is needed. But... It does not like this code:

```
^ the trait `std::convert::From<prost_codec::Error>` is not implemented for `error::GossipsubHandlerError`
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the trait `std::convert::From<std::io::Error>` is implemented for `error::GossipsubHandlerError`
    = note: required because of the requirements on the impl of `std::ops::FromResidual<std::result::Result<std::convert::Infallible, 
prost_codec::Error>>` for `std::result::Result<(), error::GossipsubHandlerError>`

```
...which makes me think it is asking me to implement a new trait. I'm not sure if that is the right way to go, or changing the Result from this encode functions, or... something else I'm completely missing. :thinking: :confused: